### ISSUE TITLE
 npm run build     Error!

### ISSUE THREAD

**Issue by yibeef** [yasp-dota/yasp#812]
I build the yasp in ubuntu15.04.
everything is right before I exec "sudo npm run build"
Error info:
   mvn -q -f java_parser/pom.xml clean install -U 
 Linux 4.2.0-16-generic
 argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "maven"
node v5.3.0
 npm  v3.5.2
yasp@6.0.0 maven: `mvn -q -f java_parser/pom.xml clean install -U`
Exit status 1

[npm-debug.txt](https://github.com/yasp-dota/yasp/files/79830/npm-debug.txt)

" Tell the author that this fails on your system:
    mvn -q -f java_parser/pom.xml clean install -U"

**Comment by WTFKr0** [comment]
Hello
Try to just run that maven command and send output.
mvn -q -f java_parser/pom.xml clean install -U

**Comment by yibeef** [comment]
Juts get the same result.

**Comment by WTFKr0** [comment]
Apperently "Exit 1" is for JAVA_HOME not set.
Try : 
echo $JAVA_HOME

**Comment by howardchung** [comment]
Check your maven install: `mvn -version`

**Comment by howardchung** [comment]
it also requires Java 8, although you could edit pom.xml to use Java 7 if you don't want to use it

**Comment by yibeef** [comment]
$ mvn -version
Apache Maven 3.3.3
Maven home: /usr/share/maven
Java version: 1.8.0_66-internal, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: zh_CN, platform encoding: UTF-8
OS name: "linux", version: "4.2.0-16-generic", arch: "amd64", family: "unix"

$ echo $JAVA_HOME
/usr/lib/jvm/java-8-openjdk-amd64

Everything looks fine but still  get the same error

**Comment by howardchung** [comment]
Remove the -q flag to get more logs:

mvn -f java_parser/pom.xml clean install -U

**Comment by yibeef** [comment]
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building stats 0.1.0
[INFO] ------------------------------------------------------------------------
Downloading: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-clean-plugin/2.5/maven-clean-plugin-2.5.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.396 s
[INFO] Finished at: 2016-01-07T18:22:23+08:00
[INFO] Final Memory: 8M/109M
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin org.apache.maven.plugins:maven-clean-plugin:2.5 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.maven.plugins:maven-clean-plugin:jar:2.5: Could not transfer artifact org.apache.maven.plugins:maven-clean-plugin:pom:2.5 from/to central (https://repo.maven.apache.org/maven2): java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException

**Comment by howardchung** [comment]
I've actually never seen that before.  You can try Googling for the error to see if others have seen it.

**Comment by likered** [comment]
Ran into the same error.

**Comment by howardchung** [comment]
I can't seem to repro this locally and Travis doesn't seem to have a problem either.

**Comment by yibeef** [comment]
can u tell me which  ubuntu version used in your yasp server?

**Comment by howardchung** [comment]
14.10 I think.

**Comment by WTFKr0** [comment]
Seems similar to : 
http://stackoverflow.com/questions/30522679/plugin-org-apache-maven-pluginsmaven-clean-plugin2-5-or-one-of-its-dependencie

**Comment by likered** [comment]
I'm using 15.10.

**Comment by howardchung** [comment]
If you have access to a 14.04 (trusty) box try it?  I tweaked init.sh and was able to bootstrap on 14.04 cloud9 devbox.

**Comment by likered** [comment]
I got 14.04 TLS box, the installations went okay. However, I couldn't just `sudo bash init.sh`. Installation breaks at `sudo apt-get -y install make g++ build-essential redis-server postgresql-9.5 openjdk-8-jdk url git maven cassandra` at `url`. There's no package named `url`.

Other than that, I had to run the commands one by one. No issues with that

**Comment by howardchung** [comment]
yeah that's a mistake.  I fixed it in a patch I should be committing soon

**Comment by likered** [comment]
I set my environment up flawlessly this time. How exactly do I access/work with the environment? Is it a local server hosting the entire website? 

I ran the dev environment (`npm run dev`) and it just gets a constant feed of API calls. So i'm assuming I keep that running?

**Comment by howardchung** [comment]
Check the wiki and read the Developer's Guide.  There are currently 12 service types.  `npm run dev` runs all of them at once.  Depending on what feature you are working on you may only need one running at once (usually web).

**Comment by likered** [comment]
Tried getting the sample data using (`wget -qO- https://github.com/yasp-dota/testfiles/raw/master/yasp.zip | gzip | psql -U yasp`), however, I run into `psql: FATAL: Peer authentication failed for user "yasp"`

Probably due to `psql -U yasp` Should I be setting my username as something else? Or was it due to some previous mis-configured step?

**Comment by howardchung** [comment]
Did you set up your db with the yasp user/profile/password change (in
sql/create.sh?)  It should be run by sudo npm run create

**Comment by likered** [comment]
Yes, I ran it and there were no errors. If I remember correctly, the output were 2 `ALTER ROLE`s and a lot of `CREATE TABLE`s & `CREATE INDEX`s. I didn't change any of the contents in sql/create.sh.

**Comment by howardchung** [comment]
Try psql postgresql://yasp:yasp@localhost/yasp

**Comment by likered** [comment]
Yeah I ran that before by itself and it seems to work fine. I get in fine

> root@likered:~/Work/yasp# psql postgresql://yasp:yasp@localhost/yasp
> psql (9.5.0)
> SSL connection (protocol: TLSv1.2, cipher: blah, bits: 256, compression: off)
> Type "help" for help.

> yasp=>

**Comment by howardchung** [comment]
in that case just sub the full string into the import command.  I'll update
the README

**Comment by howardchung** [comment]
committed.  I'll close this issue since the new init.sh seems to work but reopen/comment if there are further issues,

**Comment by likered** [comment]
> root@likered:~/Work/yasp# wget -qO- https://github.com/yasp-dota/testfiles/raw/master/yasp.zip | gzip | > psql -U postgresql://yasp:yasp@localhost/yasp
> psql: FATAL:  Peer authentication failed for user "postgresql://yasp:yasp@localhost/yasp"

**Comment by howardchung** [comment]
no -U, that flag specifies a user.  You just want to pass the connection
string directly.

**Comment by howardchung** [comment]
The dump may be out of date (I think I've changed the schema since it was
last exported).  If you only need a couple matches to dev against it may be
simplier just to request some match_ids through the request form.  `npm run
dev web,requests,parser`

**Comment by likered** [comment]
I gave the 2 given commands a try and you were right about the first one (`wget -qO- https://github.com/yasp-dota/testfiles/raw/master/yasp.zip | gzip | > psql postgresql://yasp:yasp@localhost/yasp`), it gave a huge dump of unicodes and output errors.

As for the second command (`npm run dev web,requests,parser,retriever`). I tried to request a match ID (a random one that I found that was finished recently), I left it over for the whole night running, but nothing actually happened. Was there any part I was missing? There weren't any errors in the output (Sorry, do not have the output dump with me at this time) so I assumed everything should work fine. The output dump was along the lines of requests.1 returning parsing job of <match id> is active.

**Comment by howardchung** [comment]
Okay, I'll delete that from the README since it doesn't work anyway.

You'll need to see what parser is outputting.  Is it crashing?  Stuck active means parser isn't completing the job.

**Comment by likered** [comment]
Looks like parser crashes ` Error: EACCES: permission denied, open 'output.json'`

Here's the dump when I request a match:

---------
10:26:16 PM web.1    |  2016-01-13T03:26:16.009Z - info: 107173931 visit /request
10:26:16 PM retriever.1 |  acct 0 ready, 1/1
10:26:35 PM web.1    |  2016-01-13T03:26:35.843Z - info: 107173931 visit /request_job
10:26:36 PM web.1    |  created jobId: 1
10:26:36 PM requests.1 |  2016-01-13T03:26:36.044Z - info: getData: http://api.steampowered.com/IDOTA2Match_570/GetMatchDetails/V001/?key=4F01BD68BB0D6B6733D6E4E8A6E8259A&match_id=2070192489
10:26:37 PM requests.1 |  created jobId: 3
10:26:37 PM requests.1 |  created jobId: 2
10:26:37 PM parser.1 |  2016-01-13T03:26:37.590Z - info: getData: http://localhost:5100/?key=&match_id=2070192489
10:26:38 PM web.1    |  2016-01-13T03:26:38.052Z - info: 107173931 visit /request_job?id=1
10:26:38 PM retriever.1 |  [DOTA] requesting replay 2070192489, numusers: 1, requests: 0
10:26:39 PM retriever.1 |  12 Jan 22:26:39 - Received match data for: 2070192489
10:26:39 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":0.15,"remaining":null},"speed":null,"percentage":0.003133667959335714,"size":{"total":21…
10:26:39 PM requests.1 |  waiting for parse job 2, currently in active
10:26:40 PM web.1    |  2016-01-13T03:26:40.051Z - info: 107173931 visit /request_job?id=1
10:26:40 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":1.15,"remaining":31.415},"speed":651824.347826087,"percentage":0.03531361789263256,"size…
10:26:41 PM requests.1 |  waiting for parse job 2, currently in active
10:26:42 PM web.1    |  2016-01-13T03:26:42.054Z - info: 107173931 visit /request_job?id=1
10:26:42 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":3.586,"remaining":97.376},"speed":210245.95649749026,"percentage":0.03551826405781122,"s…
10:26:43 PM requests.1 |  waiting for parse job 2, currently in active
10:26:43 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":4.585,"remaining":50.392},"speed":386103.8167938931,"percentage":0.08339830597824023,"si…
10:26:44 PM web.1    |  2016-01-13T03:26:44.052Z - info: 107173931 visit /request_job?id=1
10:26:44 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":5.585,"remaining":25.382},"speed":685477.8871978514,"percentage":0.18035592792196234,"si…
10:26:45 PM requests.1 |  waiting for parse job 2, currently in active
10:26:45 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":6.584,"remaining":15.854},"speed":946020.9599027947,"percentage":0.29342990647425277,"si…
10:26:46 PM web.1    |  2016-01-13T03:26:46.049Z - info: 107173931 visit /request_job?id=1
10:26:46 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":7.584,"remaining":9.229},"speed":1262509.7573839664,"percentage":0.45107303088602463,"si…
10:26:47 PM requests.1 |  waiting for parse job 2, currently in active
10:26:47 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":8.583,"remaining":5.542},"speed":1502809.9732028428,"percentage":0.6076548595314187,"siz…
10:26:48 PM web.1    |  2016-01-13T03:26:48.050Z - info: 107173931 visit /request_job?id=1
10:26:48 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":9.583,"remaining":2.91},"speed":1699042.2623395596,"percentage":0.7670425642352937,"size…
10:26:49 PM requests.1 |  waiting for parse job 2, currently in active
10:26:49 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":10.587,"remaining":1.349},"speed":1778363.8424482858,"percentage":0.886966724552386,"siz…
10:26:50 PM web.1    |  2016-01-13T03:26:50.050Z - info: 107173931 visit /request_job?id=1
10:26:50 PM parser.1 |  {"url":"http://replay123.valve.net/570/2070192489_1474973176.dem.bz2","state":{"time":{"ellapsed":11.586,"remaining":0.457},"speed":1762627.4814431209,"percentage":0.9620726209341532,"si…
10:26:51 PM requests.1 |  waiting for parse job 2, currently in active
10:26:51 PM parser.1 |  total time taken: 10.973
10:26:52 PM web.1    |  2016-01-13T03:26:52.051Z - info: 107173931 visit /request_job?id=1
10:26:52 PM parser.1 |  time spent on post-processing match : 1053.926ms
10:26:52 PM parser.1 |  fs.js:584
10:26:52 PM parser.1 |    return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
10:26:52 PM parser.1 |                   ^
10:26:52 PM parser.1 |  Error: EACCES: permission denied, open 'output.json'
10:26:52 PM parser.1 |      at Error (native)
10:26:52 PM parser.1 |      at Object.fs.openSync (fs.js:584:18)
10:26:52 PM parser.1 |      at Object.fs.writeFileSync (fs.js:1224:33)
10:26:52 PM parser.1 |      at /home/derek/Work/yasp/parser.js:51:16
10:26:52 PM parser.1 |      at DestroyableTransform.exit (/home/derek/Work/yasp/runParse.js:88:16)
10:26:52 PM parser.1 |      at emitNone (events.js:72:20)
10:26:52 PM parser.1 |      at DestroyableTransform.emit (events.js:166:7)
10:26:52 PM parser.1 |      at /home/derek/Work/yasp/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:965:16
10:26:52 PM parser.1 |      at nextTickCallbackWith0Args (node.js:456:9)
10:26:52 PM parser.1 |      at process._tickCallback (node.js:385:13)
10:26:52 PM parser.1 |  [nodemon] app crashed - waiting for file changes before starting...
10:26:53 PM requests.1 |  waiting for parse job 2, currently in active
10:26:54 PM web.1    |  2016-01-13T03:26:54.050Z - info: 107173931 visit /request_job?id=1
10:26:55 PM requests.1 |  waiting for parse job 2, currently in active
10:26:56 PM web.1    |  2016-01-13T03:26:56.052Z - info: 107173931 visit /request_job?id=1
10:26:57 PM requests.1 |  waiting for parse job 2, currently in active
10:26:58 PM web.1    |  2016-01-13T03:26:58.048Z - info: 107173931 visit /request_job?id=1
10:26:59 PM requests.1 |  waiting for parse job 2, currently in active
10:27:00 PM web.1    |  2016-01-13T03:27:00.049Z - info: 107173931 visit /request_job?id=1

**Comment by howardchung** [comment]
Whoops that shouldn't be there, it's for debugging and we don't want to be writing every parse to file in prod.

Fixed and pushed.

That should fix your problem, but it's weird that you can't write to your local directory.  Perhaps it's owned by root (you created it via sudo?) and you're running yasp as non-root?

**Comment by likered** [comment]
I did everything in sudo, even running yasp

Cool, let me try again

**Comment by howardchung** [comment]
You should not be doing everything in sudo.  Generally you are only
supposed to use sudo for the stuff that needs it, or else you might have
issues.

**Comment by likered** [comment]
Whoops, haha. I'm not too experienced of a linux user, so I end up assuming that it's much easier to work in my sudo shell rather than typing sudo everytime.

Let me set up my environment from scratch again to prevent any further issues that might arise in the future.
